### DISCUSSION TITLE
about backlog2

### DISCUSSION THREAD

**Discussion by Meraroo** [ANLAB-KAIST/KENSv3#68] [Project Q&A 2. Transport Layer (TCP)] [answered]
![image](https://user-images.githubusercontent.com/81629557/162490574-d1ec60f4-901d-47bd-94a7-ec028dc83706.png)
![image](https://user-images.githubusercontent.com/81629557/162490593-83f30ba0-9c89-47ff-b021-a133f1c42391.png)

제가 정확하게 이해했는지는 모르겠지만 backlog2 에서 6개의 connect 요청을 10msec 단위로 보내고, 
accept 쪽에서는 1초 후에 accept 요청 6번을 하는 것으로 이해했습니다.

이 테스트 케이스에서 backlog 가 1로 설정되어 있으므로 connect 요청은 1개만 backlog 에 남아있고 나머지는 discard 되므로
accept 가 한 번만 일어나야 하는게 아닌가요? #58 번 답변을 봤는데도 이해가 안 되어 질문드립니다.
277번째 줄의 makeTime (1, TimeUtil::SEC) 을 makeTime (1, TimeUtil::MSEC) 으로 바꾸었을때는 통과했습니다.

**Comment by Pusnow** [comment, marked as answer]
In OS, connection establishment is done like below:


1. 3-way handshaking
2. established (3-way handshaked, ESTABLISHED state in TCP, but program does not call `accept(2)`)
3. completely established (the program calls `accept(2)`)


The `backlog` in KENS the limits the number connections in the first stage.
Because there are large time intervals between `connect` requests, the number is always 0 or 1.

On the other hand, in the modern OSes (like Linux after v2.2), the `backlog` limits the number of connections in the first and second stages.
